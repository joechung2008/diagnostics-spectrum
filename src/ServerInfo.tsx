import {
  Cell,
  Column,
  Row,
  TableBody,
  TableHeader,
  TableView,
} from "@adobe/react-spectrum";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      id: "hostname",
      name: "Hostname",
      value: hostname,
    },
    {
      id: "uptime",
      name: "Uptime",
      value: uptime,
    },
    {
      id: "serverId",
      name: "Server ID",
      value: serverId,
    },
    {
      id: "deploymentId",
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      id: "nodeVersions",
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      id: "extensionSync",
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <TableView aria-label="Server Info" width="100%">
      <TableHeader>
        <Column>Name</Column>
        <Column>Value</Column>
      </TableHeader>
      <TableBody>
        {items.map((item) => (
          <Row key={item.id}>
            <Cell>{item.name}</Cell>
            <Cell>{item.value}</Cell>
          </Row>
        ))}
      </TableBody>
    </TableView>
  );
};

export default ServerInfo;
