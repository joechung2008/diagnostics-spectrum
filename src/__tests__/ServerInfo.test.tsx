import { describe, expect, it } from "vitest";
import ServerInfo from "../ServerInfo";
import { render } from "./test-utils";

describe("ServerInfo", () => {
  it("renders correctly", () => {
    const serverInfoProps = {
      deploymentId: "dep-123",
      extensionSync: { totalSyncAllCount: 5 },
      hostname: "localhost",
      nodeVersions: "v18.0.0",
      serverId: "srv-456",
      uptime: 123456,
    };
    const { container } = render(<ServerInfo {...serverInfoProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
