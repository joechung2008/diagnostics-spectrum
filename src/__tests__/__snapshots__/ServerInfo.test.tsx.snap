// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > renders correctly 1`] = `
<div
  class="_t8qIa_spectrum kDKRXa_spectrum kDKRXa_i18nFontFamily YqfL3a_spectrum--light _1DrGeG_spectrum--large XhWg9q_spectrum XhWg9q_spectrum--dark XhWg9q_spectrum--darkest XhWg9q_spectrum--large XhWg9q_spectrum--light XhWg9q_spectrum--lightest XhWg9q_spectrum--medium"
  dir="ltr"
  lang="en-US"
  style="isolation: isolate; color-scheme: light;"
>
  <span
    data-focus-scope-start="true"
    hidden=""
  />
  <div
    aria-colcount="2"
    aria-describedby=""
    aria-label="Server Info"
    aria-rowcount="7"
    class="hm005a_spectrum-Table hm005a_spectrum-Table--regular FsDyjq_react-spectrum-Table"
    data-collection="react-aria-«r1»"
    id="react-aria-«r0»"
    role="grid"
    style="width: 100%;"
    tabindex="0"
  >
    <div
      class="hm005a_spectrum-Table-headWrapper"
      role="presentation"
      style="height: 40px; overflow: hidden; position: relative; scroll-padding-inline-start: 0;"
    >
      <div
        class="spectrum-Table-head"
        role="rowgroup"
        style="position: sticky; display: inline-block; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 0px; width: 18014398509481982px; height: 40px;"
      >
        <div
          aria-rowindex="1"
          role="row"
          style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 0px; left: 0px; width: 18014398509481982px; height: 40px;"
        >
          <div
            class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
            role="presentation"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 3; contain: size layout style; top: 0px; left: 0px; width: 9007199254740991px; height: 40px;"
          >
            <div
              aria-colindex="1"
              class="hm005a_spectrum-Table-headCell FsDyjq_react-spectrum-Table-cell"
              data-collection="react-aria-«r1»"
              data-key="$.0"
              data-react-aria-pressable="true"
              id="react-aria-«r0»-$.0"
              role="columnheader"
              tabindex="-1"
            >
              <div
                class="hm005a_spectrum-Table-headCellContents"
              >
                Name
              </div>
            </div>
          </div>
          <div
            class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
            role="presentation"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 2; contain: size layout style; top: 0px; left: 9007199254740991px; width: 9007199254740991px; height: 40px;"
          >
            <div
              aria-colindex="2"
              class="hm005a_spectrum-Table-headCell FsDyjq_react-spectrum-Table-cell"
              data-collection="react-aria-«r1»"
              data-key="$.1"
              data-react-aria-pressable="true"
              id="react-aria-«r0»-$.1"
              role="columnheader"
              tabindex="-1"
            >
              <div
                class="hm005a_spectrum-Table-headCellContents"
              >
                Value
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="hm005a_spectrum-Table-body FsDyjq_react-spectrum-Table-body"
      role="rowgroup"
      style="padding: 0px; flex: 1 1 0%; scroll-padding-inline-start: 0; overflow: auto;"
      tabindex="-1"
    >
      <div
        role="presentation"
        style="width: 18014398509481982px; height: 306px; pointer-events: auto; position: relative; overflow: visible;"
      >
        <div
          role="presentation"
          style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 0px; left: 0px; width: 18014398509481982px; height: 306px;"
        >
          <div
            aria-labelledby="react-aria-«r0»-hostname-$.0"
            aria-rowindex="2"
            class="hm005a_spectrum-Table-row hm005a_spectrum-Table-row--firstRow FsDyjq_react-spectrum-Table-row"
            data-collection="react-aria-«r1»"
            data-key="hostname"
            data-react-aria-pressable="true"
            id="react-aria-«r5»"
            role="row"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 0px; left: 0px; width: 18014398509481982px; height: 51px;"
            tabindex="-1"
          >
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 0px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="1"
                aria-labelledby="react-aria-«r0»-hostname-$.0"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="hostname.0"
                role="rowheader"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«r0»-hostname-$.0"
                >
                  Hostname
                </span>
              </div>
            </div>
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 9007199254740991px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="2"
                aria-labelledby="react-aria-«r7»"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="hostname.1"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«r7»"
                >
                  localhost
                </span>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-aria-«r0»-uptime-$.0"
            aria-rowindex="3"
            class="hm005a_spectrum-Table-row FsDyjq_react-spectrum-Table-row"
            data-collection="react-aria-«r1»"
            data-key="uptime"
            data-react-aria-pressable="true"
            id="react-aria-«r8»"
            role="row"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 51px; left: 0px; width: 18014398509481982px; height: 51px;"
            tabindex="-1"
          >
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 0px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="1"
                aria-labelledby="react-aria-«r0»-uptime-$.0"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="uptime.0"
                role="rowheader"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«r0»-uptime-$.0"
                >
                  Uptime
                </span>
              </div>
            </div>
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 9007199254740991px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="2"
                aria-labelledby="react-aria-«ra»"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="uptime.1"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«ra»"
                >
                  123456
                </span>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-aria-«r0»-serverId-$.0"
            aria-rowindex="4"
            class="hm005a_spectrum-Table-row FsDyjq_react-spectrum-Table-row"
            data-collection="react-aria-«r1»"
            data-key="serverId"
            data-react-aria-pressable="true"
            id="react-aria-«rb»"
            role="row"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 102px; left: 0px; width: 18014398509481982px; height: 51px;"
            tabindex="-1"
          >
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 0px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="1"
                aria-labelledby="react-aria-«r0»-serverId-$.0"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="serverId.0"
                role="rowheader"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«r0»-serverId-$.0"
                >
                  Server ID
                </span>
              </div>
            </div>
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 9007199254740991px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="2"
                aria-labelledby="react-aria-«rd»"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="serverId.1"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«rd»"
                >
                  srv-456
                </span>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-aria-«r0»-deploymentId-$.0"
            aria-rowindex="5"
            class="hm005a_spectrum-Table-row FsDyjq_react-spectrum-Table-row"
            data-collection="react-aria-«r1»"
            data-key="deploymentId"
            data-react-aria-pressable="true"
            id="react-aria-«re»"
            role="row"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 153px; left: 0px; width: 18014398509481982px; height: 51px;"
            tabindex="-1"
          >
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 0px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="1"
                aria-labelledby="react-aria-«r0»-deploymentId-$.0"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="deploymentId.0"
                role="rowheader"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«r0»-deploymentId-$.0"
                >
                  Deployment ID
                </span>
              </div>
            </div>
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 9007199254740991px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="2"
                aria-labelledby="react-aria-«rg»"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="deploymentId.1"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«rg»"
                >
                  dep-123
                </span>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-aria-«r0»-nodeVersions-$.0"
            aria-rowindex="6"
            class="hm005a_spectrum-Table-row FsDyjq_react-spectrum-Table-row"
            data-collection="react-aria-«r1»"
            data-key="nodeVersions"
            data-react-aria-pressable="true"
            id="react-aria-«rh»"
            role="row"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 204px; left: 0px; width: 18014398509481982px; height: 51px;"
            tabindex="-1"
          >
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 0px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="1"
                aria-labelledby="react-aria-«r0»-nodeVersions-$.0"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="nodeVersions.0"
                role="rowheader"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«r0»-nodeVersions-$.0"
                >
                  Node Versions
                </span>
              </div>
            </div>
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 9007199254740991px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="2"
                aria-labelledby="react-aria-«rj»"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="nodeVersions.1"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«rj»"
                >
                  v18.0.0
                </span>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="react-aria-«r0»-extensionSync-$.0"
            aria-rowindex="7"
            class="hm005a_spectrum-Table-row hm005a_spectrum-Table-row--lastRow FsDyjq_react-spectrum-Table-row"
            data-collection="react-aria-«r1»"
            data-key="extensionSync"
            data-react-aria-pressable="true"
            id="react-aria-«rk»"
            role="row"
            style="position: absolute; overflow: visible; opacity: 1; z-index: 0; contain: size layout style; top: 255px; left: 0px; width: 18014398509481982px; height: 51px;"
            tabindex="-1"
          >
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 0px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="1"
                aria-labelledby="react-aria-«r0»-extensionSync-$.0"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="extensionSync.0"
                role="rowheader"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«r0»-extensionSync-$.0"
                >
                  Extension Sync | Total Sync All Count
                </span>
              </div>
            </div>
            <div
              class="hm005a_spectrum-Table-cellWrapper FsDyjq_react-spectrum-Table-cellWrapper"
              role="presentation"
              style="position: absolute; overflow: visible; opacity: 1; z-index: 1; contain: size layout style; top: 0px; left: 9007199254740991px; width: 9007199254740991px; height: 50px;"
            >
              <div
                aria-colindex="2"
                aria-labelledby="react-aria-«rm»"
                class="hm005a_spectrum-Table-cell FsDyjq_react-spectrum-Table-cell"
                data-collection="react-aria-«r1»"
                data-key="extensionSync.1"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="hm005a_spectrum-Table-cellContents"
                  id="react-aria-«rm»"
                >
                  5
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="hm005a_spectrum-Table-bodyResizeIndicator"
          style="left: 0px; height: 306px; display: none;"
        />
      </div>
    </div>
  </div>
  <span
    data-focus-scope-end="true"
    hidden=""
  />
</div>
`;
